include/aos/common/spaceallocator/spaceallocator.hpp

```diff
@@ -171,6 +417,99 @@ class Space : public SpaceItf {
 template <size_t cNumAllocations>
 class SpaceAllocator : public SpaceAllocatorItf {
 public:
+    /**
+     * Constructor.
+     */
+    SpaceAllocator() = default;
+    /**
+     * Destructor.
+     */
+    ~SpaceAllocator()
+    {
+        LockGuard lock {mPartitionsMutex};
+
+        mPartitions.Clear();
+    }
+
+    /**
+     * Initializes space allocator.
+     *
+     * @param path path to allocate space.
+     * @param limit limit in percents.
+     * @param remover item remover.
+     * @param platformFS platform file system.
+     * @return Error.
+     */
+    Error Init(const String& path, u_int limit, ItemRemoverItf& remover, fs::FSPlatformItf& platformFS)
+    {
+        LockGuard lock {mPartitionsMutex};
+
+        mRemover    = &remover;
+        mPlatformFS = &platformFS;
+        mPath       = path;
+
+        auto [mountPoint, errMountPoint] = mPlatformFS->GetMountPoint(path);
+        if (!errMountPoint.IsNone()) {
+            return errMountPoint;
+        }
+
+        Error err;
+
+        auto partitionIt = mPartitions.Find(mountPoint);
+        if (partitionIt == mPartitions.end()) {
+            if (Tie(mPartition, err) = NewPartition(mountPoint); !err.IsNone()) {
+                return err;
+            }
+
+            if (err = mPartitions.TryEmplace(mountPoint, mPartition); !err.IsNone()) {
+                return err;
+            }
+        } else {
+            mPartition = partitionIt->mSecond;
+        }
+
+        mPartition->mAllocatorCount++;
+
+        if (limit != 0) {
+            if (err = mPartition->AddLimit(limit); !err.IsNone()) {
+                return err;
+            }
+
+            mSizeLimit = mPartition->mTotalSize * mPartition->mLimit / 100;
+        }
+
+        return ErrorEnum::eNone;
+    }
+
+    /**
+     * Closes space allocator.
+     *
+     * @return Error.
+     */
+    Error Close()
+    {
+        LockGuard lock {mPartitionsMutex};
+
+        Error err;
+
+        if (auto errRemovePartLimit = mPartition->RemoveLimit(mPartition->mLimit); !errRemovePartLimit.IsNone()) {
+            err = errRemovePartLimit;
+        }
+
+        mPartition->mAllocatorCount--;
+
+        if (mPartition->mAllocatorCount != 0) {
+            return err;
+        }
+
+        if (auto errRemovePart = mPartitions.Remove(mPartition->mMountPoint);
```

Comment:
We should have something like this:

```c++
static inline StaticMap<String, Partition, cMaxNumPartitions> mPartitions;
```

And no need to remove partition from map